stores/yaml/store.go

```diff
@@ -118,45 +122,89 @@ func (store *Store) LoadEncryptedFile(in []byte) (sops.Tree, error) {
 		return sops.Tree{}, err
 	}
 	// After that, we load the whole file into a map.
-	var data yaml.MapSlice
-	if err := (yaml.CommentUnmarshaler{}).Unmarshal(in, &data); err != nil {
-		return sops.Tree{}, fmt.Errorf("Error unmarshaling input YAML: %s", err)
-	}
 	// Discard metadata, as we already loaded it.
-	for i, item := range data {
+	for i, item := range data[0] {
 		if item.Key == "sops" {
-			data = append(data[:i], data[i+1:]...)
+			data[0] = append(data[0][:i], data[0][i+1:]...)
 		}
 	}
+
+	var branches sops.TreeBranches
+	for _, doc := range data {
```

Comment:
I feel like either this or the previous for loop are redundant, no? We only store the metadata in the first document, and we're first removing the metadata entries from the first tree and then from *all* trees.